### ISSUE TITLE
Support `.clang-format-ignore` for `ninja clang-format` target

### ISSUE THREAD

**Issue by XVilka** [mesonbuild/meson#7271]
**Describe the bug**
Currently `ninja clang-format` ignores the `.clang-format-ignore` file by running the clang-format against those files listed in the ignore file.

**To Reproduce**
```sh
git clone https://github.com/XVilka/radare2
cd radare2 && git checkout clang-format
meson build
cd build && ninja clang-format
```
See https://github.com/radareorg/radare2/pull/17015 for more context

See the example of such file at https://github.com/mozilla/gecko-dev/blob/master/.clang-format-ignore

**Expected behavior**
Runs `clang-format` **ignoring** the files listed in `.clang-format-ignore`.

**system parameters**
* Native
* Fedora 32 x86_64
* Python 3.8.0
* Meson 0.54.2
* Ninja 1.10.0
* clang-format version 10.0.0 (Fedora 10.0.0-1.fc32)

**Comment by TheQwertiest** [comment]
It should be noted that `.clang-format-ignore` is not a built-in `clang-format` functionality. I.e. it's just an arbitrary file that must be manually parsed (and acted upon) before `clang-format` invocation.

**Comment by XVilka** [comment]
There is a request to support it in Visual Studio: https://developercommunity.visualstudio.com/idea/635721/support-clang-format-ignore-format.html
Moreover, seem that Mozilla uses it too: https://searchfox.org/mozilla-central/source/.clang-format-ignore

There is also a script to handle this already: https://github.com/Sarcasm/run-clang-format

**Comment by gregsskyles** [comment]
So isn't this a clang-format enhancement request, rather than a meson bug?
Each user of clang-format needs to implement support (in a different and ultimately likely incompatible fashion) for this file format?
Or are you saying meson isn't passing some parameter in the correct way to clang-format?

**Comment by TheQwertiest** [comment]
> Each user of clang-format needs to implement support (in a different and ultimately likely incompatible fashion) for this file format?
Yes, this.

**Comment by detly** [comment]
> So isn't this a clang-format enhancement request, rather than a meson bug?

`clang-format` doesn't have out-of-the-box support for recursing through a project and updating files in place. It has the *ability* to do that, when supplied with the appropriate paths and flags, but it doesn't do that by itself, so the concept of `clang-format` having an ignore file seems a bit strange. It is Meson that creates the target to apply `clang-format` recursively to a project, so IMO it should be Meson that also provides the capability to not add certain paths to the target that it generates.

A very limited workaround is to put the following in `clang-format` in the directory in question:

```text
DisableFormat: true
SortIncludes: false
```

However, this won't work for Git submodules. There is no way to have Meson's automatically generated target ignore Git submodules as far as I can see.

**Comment by XVilka** [comment]
So we opted for our own short script for this task, just in case someone will need an interim solution: https://github.com/rizinorg/rizin/blob/dev/sys/clang-format.py

**Comment by eli-schwartz** [comment]
This was implemented in #8503
